src/OpenTelemetry.Instrumentation.Wcf/Implementation/ClientChannelInstrumentation.cs

```diff
@@ -82,19 +82,25 @@ public static RequestTelemetryState BeforeSendRequest(Message request, Uri? remo
         };
     }
 
-    public static void AfterRequestCompleted(Message? reply, RequestTelemetryState? state)
+    public static void AfterRequestCompleted(Message? reply, RequestTelemetryState? state, Exception? exception = null)
     {
         Guard.ThrowIfNull(state);
-
         state.SuppressionScope?.Dispose();
-
         if (state.Activity is Activity activity)
         {
             if (activity.IsAllDataRequested)
             {
                 if (reply == null || reply.IsFault)
                 {
                     activity.SetStatus(Status.Error);
+
+                    if (WcfInstrumentationActivitySource.Options!.RecordException)
+                    {
+                        if (exception != null)
+                        {
+                            activity.RecordException(exception);
```

Comment:
This extension is being marked `Obsolete` in `1.10.0`. We should call [Activity.AddException](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.addexception) here. To do that you'll need to wait for the repo to bump to 1.10.0 or you can switch your OTel reference to use `OpenTelemetryCoreLatestPrereleaseVersion`.

Reply:
I think a better option would be to change it after the bump, along with other libraries where we have record exception

Reply:
Bump done. Method updated.